packages/network/src/v2/keyTupleToEntityID.ts

```diff
@@ -1,12 +1,13 @@
 import { Entity } from "@latticexyz/recs";
-import { normalizeEntityID } from "../utils";
+import { toHex, pad, isHex, Hex } from "viem";
 import { SingletonID } from "../workers";
 
 // TODO: revisit key tuple format?
 export function keyTupleToEntityID(keyTuple: any[]): Entity {
   // v2 uses an empty key tuple as the singleton ID, so we'll return the corresponding v1 singleton entity ID to normalize this for now
   if (keyTuple.length === 0) {
-    return SingletonID;
+    return pad(SingletonID as Hex, { size: 32 }) as Entity;
```

Comment:
It'll take a little extra work (mostly loading states and in templates + tutorials) but I think we can probably move this to `return ""` and nix the `singletonEntity` stuff. cc @ludns

Reply:
since I'm here and already modifying `playerEntity`, should I just rip out `singletonEntity` too?

Reply:
I'd rip all of those, unless we are thinking of moving straight to refactoring all of this and killing; and making this a non breaking change that just fixes things instead of cleaning the API